src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583ReplySender.java

```diff
@@ -61,13 +66,29 @@ public ISO8583ReplySender(Socket connection) {
     public void sendBack(MessageContext messageContext) {
         String responseMessage = null;
         try {
-            ISOPackager packager = ISO8583PackagerFactory.getPackager();
+            ISOBasePackager packager = ISO8583PackagerFactory.getPackagerWithParams(params);
             Properties properties = getPropertiesFile();
             //Retrieve the SOAP envelope from the MessageContext
             SOAPEnvelope soapEnvelope = messageContext.getEnvelope();
-            OMElement getElements = soapEnvelope.getBody().getFirstElement();
+            OMElement getElements = null;
+            if (isProxy()) {
```

Comment:
what is the isProxy implementation?

Reply:
isProxy defines whether this inbound is acting as a proxy for another backend service or processing the message itself. If the inbound endpoint act as a proxy to another service, pack the ISO message without change any field